ui/raidboss/emulator/data/network_log_converter/LineEvent.ts

```diff
@@ -61,3 +61,70 @@ export default class LineEvent {
       ).toString(16), 16);
   }
 }
+
+// Type guards for these interfaces require their own descriptor property
+// because we don't want every line event with an id/name
+// to update combatant state, for example
+export interface LineEventSource {
+  readonly isSource: true;
+  readonly id: string;
+  readonly name: string;
+  readonly x?: number;
+  readonly y?: number;
+  readonly z?: number;
+  readonly heading?: number;
+  readonly targetable?: boolean;
+  readonly hp?: number;
+  readonly maxHp?: number;
+  readonly mp?: number;
+  readonly maxMp?: number;
+}
+
+export const isLineEventSource =
+  (line: LineEventSource | LineEvent): line is LineEventSource => {
```

Comment:
```suggestion
  (line: LineEvent): line is LineEventSource => {
```

What about this change along with making `LineEventSource` / `LineEventTarget` / etc all extend `LineEvent`?